heudiconv/tests/test_heuristics.py

```diff
@@ -36,6 +38,9 @@ def test_smoke_convertall(tmpdir):
     args.extend(['-f', 'convertall'])
     runner(args)
 
+    # ensure we are not leaving lingering temporary directories
+    assert not list(Path(tempfile.gettempdir()).glob('heudiconv*'))
```

Comment:
note: wouldn't be safe (fail) in case of parallel testing, not that we are doing it now but someone could with e.g. py.test xdist

Reply:
and on the other hand - since  we do not use `heudiconv` prefix consistently across all uses - it might miss some temp dir to not be cleaned up.

I will push adding consistent `heudiconv` prefix (possibly changing names) to all temp dirs invocations.  Not that it how it was, but I think it would be good to have all temp dirs with consistent prefix.

Reply:
good point - do you have any recommendations on a cleaner way to test for linger tmpdirs?

Reply:
patch `TMPDIR` env var for this run, point to some directory... at little fragile since AFAIK we would not know if underlying code actually uses it, so would need an extra check (would be a useful test on its own IMHO) to test that `tmpdirs.add_tmpdir` and `tempfile.TemporaryDirectory(prefix='heudiconv-')` create a directory there.

also note -- `runner` AFAIK runs in the same process, doesn't it? so atexit wouldn't work out, so I guess we would need manual `.cleanup` here?